examples/qgc_proxy_px4/qgc_proxy_px4.cpp

```diff
@@ -0,0 +1,183 @@
+        }
+
+        std::shared_ptr<MavlinkPassthrough> passthrough;
+        {
+            std::lock_guard<std::mutex> lock(passthrough_map_mutex);
+            passthrough = passthrough_map[target_name];
+        }
+
+        if (!passthrough) {
+            log_message("ERROR", "Passthrough not found for target: " + target_name);
+            continue;
+        }
+
+        auto result = passthrough->queue_message([&](MavlinkAddress, uint8_t) { return message; });
+        if (result != MavlinkPassthrough::Result::Success) {
+            log_message("ERROR", "Failed to forward message ID: " + std::to_string(message.msgid) + " to " + target_name);
+        }
+    }
+}
+
+// Waits for a system to connect
+std::optional<std::shared_ptr<System>> wait_for_system(Mavsdk& mavsdk) {
+    for (int i = 0; i < TIMEOUT_SECONDS; ++i) {
+        if (!mavsdk.systems().empty()) {
+            return mavsdk.systems().front();
+        }
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+    return std::nullopt;
+}
+
+int main(int argc, char** argv) {
+    signal(SIGINT, signal_handler);
+
+    if (argc != 3) {
+        log_message("ERROR", "Usage: " + std::string(argv[0]) + " <url_qgc> <url_px4>");
+        return 1;
+    }
+
+    Mavsdk mavsdk_px4(Mavsdk::Configuration(PROXY_SYSID_PX4, 1, true));
+    Mavsdk mavsdk_qgc(Mavsdk::Configuration(PROXY_SYSID_QGC, 1, true));
+
+    auto px4_conn_result = mavsdk_px4.add_any_connection(argv[2]);
+    auto qgc_conn_result = mavsdk_qgc.add_any_connection(argv[1]);
+
+    if (px4_conn_result != ConnectionResult::Success || qgc_conn_result != ConnectionResult::Success) {
+        log_message("ERROR", "Failed to connect to the provided URLs.");
+        return 1;
+    }
+
+    log_message("INFO", "Waiting for systems to connect...");
+
+    intercept_outgoing_messages(mavsdk_px4, PROXY_SYSID_PX4, PROXY_SYSID_QGC);
+    //intercept_outgoing_messages(mavsdk_qgc, PROXY_SYSID_PX4, PROXY_SYSID_QGC);
+
+    auto px4_system = wait_for_system(mavsdk_px4);
+    auto qgc_system = wait_for_system(mavsdk_qgc);
+
+    if (!px4_system || !qgc_system) {
+        log_message("ERROR", "Failed to connect to one or both systems within the timeout.");
+        return 1;
+    }
+
+    {
+        std::lock_guard<std::mutex> lock(passthrough_map_mutex);
+        passthrough_map[PX4_NAME] = std::make_shared<MavlinkPassthrough>(px4_system.value());
+        passthrough_map[QGC_NAME] = std::make_shared<MavlinkPassthrough>(qgc_system.value());
+    }
+
+    message_queues[PX4_NAME] = std::queue<mavlink_message_t>();
+    message_queues[QGC_NAME] = std::queue<mavlink_message_t>();
+    queue_mutexes[PX4_NAME] = std::make_unique<std::mutex>();
+    queue_mutexes[QGC_NAME] = std::make_unique<std::mutex>();
+    queue_cvs[PX4_NAME] = std::make_unique<std::condition_variable>();
+    queue_cvs[QGC_NAME] = std::make_unique<std::condition_variable>();
+
+    for (uint16_t msg_id = 0; msg_id < 500; ++msg_id) {
```

Comment:
Are you sure that you want to subscribe 500 times? Shouldn't you subscribe just once for each? 🤔

Reply:
Hm, and why 500? That seems somewhat arbitrary, as message IDs go up to 43000, with mostly gaps in-between.

Reply:
If we want to enable this, should we maybe have something like a `subscribe_all_messages` in passthrough?

Reply:
As a Proxy, the correct thing to do is to subscribe to all message IDs. So a `subscribe_all_messages` might help. Another option is to create a list of message ID constants. I even went as far as manually importing the ID constants into a list. However, I'm open to suggestions.

Reply:
Oh, I see the problem.

I see that `Mavsdk::intercept_messages_async` offers that but MavlinkPassthrough does not. It would probably be a good idea to add that.